era_5g_interface/channels.py

```diff
@@ -0,0 +1,294 @@
+import logging
+import time
+from abc import ABC
+from dataclasses import dataclass
+from enum import Enum
+from typing import Any, Callable, Dict, List, Optional, Union
+
+import cv2
+import numpy as np
+import socketio
+
+from era_5g_interface.exceptions import UnknownChannelTypeUsed
+from era_5g_interface.h264_decoder import H264Decoder, H264DecoderError
+from era_5g_interface.h264_encoder import H264Encoder, H264EncoderError
+
+logger = logging.getLogger(__name__)
+
+DATA_NAMESPACE = str("/data")
+DATA_ERROR_EVENT = str("data_error")
+
+CONTROL_NAMESPACE = str("/control")
+COMMAND_EVENT = str("command")
+COMMAND_ERROR_EVENT = str("command_error")
+COMMAND_RESULT_EVENT = str("command_result")
+
+
+class ChannelType(Enum):
+    """Channel type dataclass."""
+
+    JSON = 1
+    JPEG = 2
+    H264 = 3
+
+
+@dataclass
+class CallbackInfoClient:
+    """Callback info dataclass."""
+
+    type: ChannelType  # Channel type.
+    callback: Callable[[dict], None]  # Custom callback.
+    error_event: str = DATA_ERROR_EVENT  # Custom error event name.
+
+
+@dataclass
+class CallbackInfoServer:
```

Comment:
What is the point of having two exactly the same classes?

Reply:
Due to different callback type annotations and exclusive use in individual classes.

Reply:
I see, I missed that the annotation is different.